packages/modular-scripts/src/rename.ts

```diff
@@ -0,0 +1,150 @@
+  );
+
+  const packageJson = (await fs.readJson(
+    newPackageJsonLocation,
+  )) as PackageJson;
+
+  logger.log(`Changing package name ${packageJson.name} → ${newPackageName}`);
+  packageJson.name = newPackageName;
+
+  await fs.writeJson(newPackageJsonLocation, packageJson, { spaces: 2 });
+
+  // Search for packages that depend on the renamed package
+  type PackageDepInfo = { packageData: WorkSpaceRecord; json: PackageJson };
+
+  logger.log(`Searching for packages depending on ${oldPackageName}`);
+  const dependingPackages: PackageDepInfo[] = (
+    await Promise.all(
+      workspace
+        .filter(([packageName]) => packageName !== oldPackageName)
+        .map(async ([_, packageData]) => ({
+          packageData,
+          json: (await fs.readJson(
+            path.join(getModularRoot(), packageData.location, './package.json'),
+          )) as PackageJson,
+        })),
+    )
+  ).filter((pkgJsonInfo) =>
+    dependencyTypes.some(
+      (depField) => pkgJsonInfo.json[depField]?.[oldPackageName],
+    ),
+  );
+
+  logger.log(`Renaming dependencies in depending packages`);
+  await Promise.all(
+    dependingPackages.map(async (pkg) => {
+      let modified = false;
+      dependencyTypes.forEach((depField) => {
+        // This needs to be explicitly assigned, otherwise Typescript doesn't understand it's defined
+        const depObject = pkg.json[depField];
+        if (depObject?.[oldPackageName]) {
+          logger.debug(
+            `Renaming dependency ${oldPackageName} → ${newPackageName} in ${depField} / ${pkg.packageData.location}`,
+          );
+          depObject[newPackageName] = depObject[oldPackageName];
+          delete pkg.json[depField]?.[oldPackageName];
+          modified = true;
+        }
+      });
+      if (modified) {
+        await fs.writeJson(
+          path.join(
+            getModularRoot(),
+            pkg.packageData.location,
+            './package.json',
+          ),
+          pkg.json,
+          { spaces: 2 },
+        );
+      }
+    }),
+  );
+
+  logger.log(`Rewriting imports in depending packages`);
+  await Promise.all(
+    dependingPackages.map(async (pkg) => {
+      const project = new Project();
+      project.addSourceFilesAtPaths(
+        path.join(
+          getModularRoot(),
+          pkg.packageData.location,
+          'src/**/*{.d.ts,.ts,.js,.jsx,.tsx}',
+        ),
+      );
+      const sourceFiles = project.getSourceFiles();
+
+      sourceFiles.forEach((sourceFile) => {
+        const imports = sourceFile.getImportDeclarations();
+        imports.forEach((importDeclaration) => {
+          if (importDeclaration.getModuleSpecifierValue() === oldPackageName) {
+            logger.debug(
+              `Rewriting \`${importDeclaration.getText()}\` in ${sourceFile.getFilePath()}`,
+            );
+            importDeclaration.setModuleSpecifier(newPackageName);
+          }
+        });
+      });
+      await project.save();
+    }),
+  );
```

Comment:
If you change the package name in the package.json, you'll need to do a `execa.sync('yarnpkg')` to update the the name of the symlink in node_modules

Reply:
Done